cxx/distributions/beta_bernoulli.cc

```diff
@@ -49,7 +49,11 @@ void BetaBernoulli::transition_hyperparameters(std::mt19937* prng) {
       }
     }
   }
-  int i = sample_from_logps(logps, prng);
-  alpha = hypers[i].first;
-  beta = hypers[i].second;
+  if (logps.empty()) {
+    printf("Warning! All hyperparamters for BetaBernoulli give nans!\n");
```

Comment:
So I think this is the behavior we want because it fulfills the implicit contract of transition_hyperparameters, which I take to be "decrease the logp_score by changing the hyperparameters, if you can".  If all the evaluated hyperparamter values give nans, well then you can't.

It's theoretically possible that the nans could go away when more data is incorporated, but I don't think it is likely.  A more likely possibility is that the nans go away when the data is reclustered, which is a good reason not to exit.

I'm not sure what it would mean here to let the nans propagate.  We aren't doing an early exit because we saw some nans, we are doing an early exit because the logps vector is empty and we can't  access an item from an empty vector.

Fixed the typo.